services/chainstorage/src/main/java/tech/pegasys/teku/services/chainstorage/StorageService.java

```diff
@@ -271,9 +271,12 @@ public Optional<StatePruner> getStatePruner() {
 
   @Override
   protected SafeFuture<?> doStop() {
-    return blockPruner
-        .map(BlockPruner::stop)
-        .orElseGet(() -> SafeFuture.completedFuture(null))
+    return SafeFuture.allOf(
+            blockPruner.map(BlockPruner::stop).orElseGet(() -> SafeFuture.completedFuture(null)),
+            blobsPruner
+                .map(BlobSidecarPruner::stop)
+                .orElseGet(() -> SafeFuture.completedFuture(null)),
+            statePruner.map(StatePruner::stop).orElseGet(() -> SafeFuture.completedFuture(null)))
```

Comment:
oh.. I just opened #9054 with this fix

Reply:
no worries, a fix is better than no fix!